arcgis-ios-sdk-samples/Maps/Browse building floors/BrowseBuildingFloorsViewController.swift

```diff
@@ -0,0 +1,115 @@
+// Copyright 2021 Esri
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//   https://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+import UIKit
+import ArcGIS
+
+class BrowseBuildingFloorsViewController: UIViewController {
+    // MARK: Storyboard views
+    
+    /// The map view managed by the view controller.
+    @IBOutlet var mapView: AGSMapView! {
+        didSet {
+            mapView.map = makeMap()
+        }
+    }
+    /// The floor level picker view.
+    @IBOutlet var floorLevelPickerView: UIPickerView!
+    
+    // MARK: Properties
+    
+    /// The levels of the floor-aware web map, with an addition of `nil`.
+    var levels: [AGSFloorLevel?] = [nil]
+    
+    // MARK: Methods
+    
+    /// Set the selected level to visible and all other levels to invisible.
+    /// - Parameter selectedLevel: The `AGSFloorLevel` selected.
+    func setVisibleFloorLevel(_ selectedLevel: AGSFloorLevel?) {
+        levels.forEach { level in
+            level?.isVisible = level == selectedLevel
+        }
```

Comment:
I know that our floor manager only has three levels, but users may have significantly more. Iterating through all of them and update the `isVisible` property of all of them is highly inefficient. Instead, it would be better to store the selected floor and use that to update the `isVisible` property of the previously selected floor and the newly selected floor:

```swift
var selectedFloorLevel: AGSFloorLevel? {
    didSet {
        oldValue?.isVisible = false
        selectedFloorLevel?.isVisible = true
        
        ... // Update picker view selection.
    }
}
```

Reply:
Great suggestion!